src/client/app/components/ThreeDPillComponent.tsx

```diff
@@ -65,13 +95,12 @@ export default function ThreeDPillComponent() {
 	);
 
 	// Method Generates Reactstrap Pill Badges for selected meters or groups
-	const populatePills = (meterOrGroupPillData: MeterOrGroupPill[]) => {
-		return meterOrGroupPillData.map(pillData => {
+	const populatePills = (pillDataArray: MeterOrGroupPill[]) => {
+		return pillDataArray.map(pillData => {
 			// retrieve data from appropriate state slice .meters or .group
-			const meterOrGroupName = pillData.meterOrGroup === MeterOrGroup.meters ?
-				meterDataById[pillData.meterOrGroupID]?.identifier
-				:
-				groupDataById[pillData.meterOrGroupID]?.name;
+			const meterOrGroupName = pillData.meterOrGroup === MeterOrGroup.meters
```

Comment:
I have not looked into this or if this is the right place but can the superscript m & g easily be added to the pill name similar to the Data Source?